### PR TITLE
Implement Creator Token Factory

### PR DESCRIPTION
**Pull request by apbendi**
Creates a CreatorTokenFactory with a deploy method that takes a configuration
struct and uses it to deploy a bonding curve contract and creator token
contract that uses said bonding curve.

The factory is integrated into the pre-existing Showtime Verifier infrastructure.
It requires the deployment be attested to by a signature from a whitelisted
verifier address before it can be deployed. The factory contract has helper
methods used to encode and generate the digest that will be signed by the verifier.

The tests for the token factory fork the Base mainnet in order to interact with
the existing onchain verifier infrastructure.

### CODE REVIEW

File: src/CreatorTokenFactory.sol [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+  bytes32 public constant DEPLOY_TYPE_HASH = keccak256(DEPLOY_TYPE);
+
+  IShowtimeVerifier public immutable VERIFIER;
+
+  // TODO: dev comment explaining this is the domain separator from the verifier that technically
+  // can change if chainId or
+  // the verifier contract address change (latter can't happen if we keep the verifier hardcoded)
+  bytes32 private constant DOMAIN_SEPARATOR =
+    0x0c3be79fa914a526eb7e8ec47178f2ca6f81e90fb14c3e335600dc3ec6fa1a6f;
```
Comment (apbendi) [inline src/CreatorTokenFactory.sol:45]: TODO: Investigate whether this changes between base mainnet & base testnet. If so, we need to add it as an immutable var configured at deployment.

File: test/CreatorTokenFactory.t.sol [hunk 2]
```diff
@@ -0,0 +1,253 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.20;
+
+import {Test, console2} from "forge-std/Test.sol";
+import {CreatorToken} from "src/CreatorToken.sol";
+import {CTBondingCurve} from "src/CTBondingCurve.sol";
+import {IERC20, ERC20} from "openzeppelin/token/ERC20/ERC20.sol";
+import {CTBondingCurve} from "src/CTBondingCurve.sol";
```
Comment (garyghayrat) [inline test/CreatorTokenFactory.t.sol:8]: Duplicate import. First import on line 6.

File: env.example [hunk 3]
```diff
@@ -0,0 +1 @@
+BASE_RPC_URL=
```
Comment (garyghayrat) [inline env.example:1]: forgot the dot in the file name? i.e. `.env.exmaple`

**Review by garyghayrat — APPROVED** [review]
2 small nits but otherwise looks good to me.

File: test/CreatorTokenFactory.t.sol (hunk 2 above)
Comment (apbendi) [reply test/CreatorTokenFactory.t.sol:8]: fixed in 029eeda4404b3c94895a3a574b80d4c216db738a

File: env.example (hunk 3 above)
Comment (apbendi) [reply env.example:1]: Fixed in c3ec33557797777563f4854696fcdc94c4e4756d

**Pull request merged**